service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOController.java

```diff
@@ -10,33 +10,53 @@
 import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
 import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.model.Direction;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.DirectionHelperService;
 import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
 
+import java.util.List;
 import java.util.Map;
 
+import static java.util.Objects.isNull;
+
 @Api
 @RestController
 @RequestMapping("/callback/draft-cmo")
 public class DraftCMOController {
     private final ObjectMapper mapper;
     private final DraftCMOService draftCMOService;
+    private final DirectionHelperService directionHelperService;
 
     @Autowired
-    public DraftCMOController(ObjectMapper mapper, DraftCMOService draftCMOService) {
+    public DraftCMOController(ObjectMapper mapper,
+                              DraftCMOService draftCMOService,
+                              DirectionHelperService directionHelperService) {
         this.mapper = mapper;
         this.draftCMOService = draftCMOService;
+        this.directionHelperService = directionHelperService;
     }
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
-        Map<String, Object> caseData = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        // Resetting allParties - could be pre-populated via SDO
+        caseDetails.getData().remove("allParties");
 
-        caseData.put("cmoHearingDateList", draftCMOService.getHearingDateDynamicList(caseDetails));
+        if (!isNull(caseData.getCaseManagementOrder())) {
+            Map<String, List<Element<Direction>>> directions = directionHelperService.sortDirectionsByAssignee(
+                caseData.getCaseManagementOrder().getDirections());
+            directions.forEach(caseDetails.getData()::put);
+        }
```

Comment:
For simplicity (and easier flow for now) wouldn't it be enough if we only extracted the allParties directio nfrom the order?

Reply:
For now yes but moving forward we are going to get more directions, what do you think?

Reply:
Might be good for convenience would just clarify in the comment that UI uses only allParties for now.